awebox/viz/variables.py

```diff
@@ -474,9 +263,115 @@ def plot_algebraic_variables(plot_dict, cosmetics, fig_name):
+        plt.ticklabel_format(style='sci', axis='x', scilimits=(0, 3))
+        plt.suptitle(fig_name)
+
+    else:
+        message = 'a request to plot variables of type ' + var_type + ' passed an empty list of variable-names.' \
+                                                                      ' as a result, the request was ignored.'
+        awelogger.logger.warning(message)
+
+    return None
+
+def plot_indiv_variable(ax, plot_dict, cosmetics, var_type, var_name):
+
+    variables_dict = plot_dict['variables_dict']
+    tgrid_ip = plot_dict['time_grids']['ip']
+
+    ax = plt.axes(ax)
+    for jdx in range(variables_dict[var_type][var_name].shape[0]):
+        p = plt.plot(tgrid_ip, plot_dict[var_type][var_name][jdx])
+        if cosmetics['plot_bounds']:
+            tools.plot_bounds(plot_dict, var_type, var_name, jdx, tgrid_ip, p)
+        if cosmetics['plot_ref']:
+            plt.plot(plot_dict['time_grids']['ref']['ip'], plot_dict['ref'][var_type][var_name][jdx],
+                     linestyle='--', color=p[-1].get_color())
+
+    plt.title(var_name)
+    ax.tick_params(axis='both', which='major')
+
+    return None
+
+def plot_indiv_integral_variable(ax, plot_dict, cosmetics, var_name):
+
+    tgrid_ip = plot_dict['time_grids']['ip']
+
+    ax = plt.axes(ax)
+
+    if plot_dict['discretization'] == 'multiple_shooting':
+        out_values, tgrid_out = tools.merge_integral_output_values(plot_dict['integral_outputs_final'], var_name, plot_dict,
+                                                                   cosmetics)
+        p = plt.plot(tgrid_out, out_values)
+    else:
+        p = plt.plot(tgrid_ip, plot_dict['integral_outputs'][var_name][0])
+
+    plt.title(var_name)
+    ax.tick_params(axis='both', which='major')
+
+    return None
+
+def setup_fig_and_axes(variables_to_plot, integral_variables_to_plot, fig_num=None):
+
+    counter = len(variables_to_plot) + len(integral_variables_to_plot)
+
+    if counter == 1:
+        plot_table_r = 1
+        plot_table_c = 1
+    elif np.mod(counter, 3) == 0:
+        plot_table_r = 3
+        plot_table_c = int(counter / plot_table_r)
+    elif np.mod(counter, 4) == 0:
+        plot_table_r = 4
+        plot_table_c = int(counter / plot_table_r)
+    else:
+        plot_table_r = 3
+        plot_table_c = int(np.ceil(np.float(counter) / np.float(plot_table_r)))
+
+    # create new figure if desired
+    if fig_num is not None:
+        fig = plt.figure(num = fig_num)
+        axes = fig.axes
+        if len(axes) == 0: # if figure does not exist yet
+            fig, axes = plt.subplots(num = fig_num, nrows = plot_table_r, ncols = plot_table_c)
+
+    else:
+        fig, axes = plt.subplots(nrows = plot_table_r, ncols = plot_table_c)
+
+    # make vertical column array or list of all axes
+    if type(axes) == np.ndarray:
+        axes = axes.reshape(plot_table_r*plot_table_c,)
+    elif type(axes) is not list:
+        axes = [axes]
+
+    return fig, axes
```

Comment:
Thanks for cleaning up!

Reply:
sure. :) hope that i didn't mess anything up here...